source/MRViewer/ImGuiMenu.cpp

```diff
@@ -385,4 +423,168 @@ ImGuiContext* ImGuiMenu::getCurrentContext() const
+        ImGui::Separator();
+        for ( int i = 0; i<int( Viewer::EventType::Count ); ++i )
+            ImGui::Text( "%s: %zu", cEventCounterNames[i], viewer->getEventsCount( Viewer::EventType( i ) ) );
+        ImGui::Separator();
+        auto prevFrameTime = viewer->getPrevFrameDrawTimeMillisec();
+        if ( prevFrameTime > frameTimeMillisecThreshold_ )
+            ImGui::TextColored( ImVec4( 1.0f, 0.3f, 0.3f, 1.0f ), "Previous frame time: %lld ms", prevFrameTime );
+        else
+            ImGui::Text( "Previous frame time: %lld ms", prevFrameTime );
+        ImGui::Text( "Total frames: %zu", viewer->getTotalFrames() );
+        ImGui::Text( "Swapped frames: %zu", viewer->getSwappedFrames() );
+        ImGui::Text( "FPS: %zu", viewer->getFPS() );
+
+        if ( ImGui::Button( "Reset", ImVec2( -1, 0 ) ) )
+        {
+            viewer->resetAllCounters();
+        }
+        if ( ImGui::Button( "Print time to console", ImVec2( -1, 0 ) ) )
+        {
+            printTimingTreeAndStop();
+        }
+        ImGui::End();
+    }
+
+    if ( showRenameModal_ )
+    {
+        showRenameModal_ = false;
+        ImGui::OpenPopup( "Rename object" );
+    }
+
+    if ( ImGui::BeginModalNoAnimation( "Rename object", nullptr,
+        ImGuiWindowFlags_NoResize | ImGuiWindowFlags_AlwaysAutoResize ) )
+    {
+        auto obj = getAllObjectsInTree( &SceneRoot::get(), ObjectSelectivityType::Selected ).front();
+        if ( !obj )
+        {
+            ImGui::CloseCurrentPopup();
+        }
+        if ( ImGui::IsWindowAppearing() )
+            ImGui::SetKeyboardFocusHere();
+        ImGui::InputText( "Name", renameBuffer_, ImGuiInputTextFlags_AutoSelectAll );
+
+        float w = ImGui::GetContentRegionAvail().x;
+        float p = ImGui::GetStyle().FramePadding.x;
+        if ( ImGui::Button( "Ok", ImVec2( ( w - p ) / 2.f, 0 ) ) || ImGui::GetIO().KeysDownDuration[GLFW_KEY_ENTER] == 0.0f )
+        {
+            AppendHistory( std::make_shared<ChangeNameAction>( "Rename object", obj ) );
+            obj->setName( renameBuffer_ );
+            ImGui::CloseCurrentPopup();
+        }
+        ImGui::SameLine( 0, p );
+        if ( ImGui::Button( "Cancel", ImVec2( ( w - p ) / 2.f, 0 ) ) || ImGui::GetIO().KeysDownDuration[GLFW_KEY_ESCAPE] == 0.0f )
+        {
+            ImGui::CloseCurrentPopup();
+        }
+
+        if ( ImGui::IsMouseClicked( 0 ) && !( ImGui::IsAnyItemHovered() || ImGui::IsWindowHovered( ImGuiHoveredFlags_AnyWindow ) ) )
+        {
+            ImGui::CloseCurrentPopup();
+        }
+
+        ImGui::EndPopup();
+    }
+
+    auto bgBackUp = ImGui::GetStyle().Colors[ImGuiCol_ModalWindowDimBg];
+
+    if ( !storedError_.empty() && !ImGui::IsPopupOpen( " Error##modal" ) )
+    {
+        ImGui::GetStyle().Colors[ImGuiCol_ModalWindowDimBg] = ImVec4( 1, 0.125f, 0.125f, bgBackUp.w );
+        ImGui::OpenPopup( " Error##modal" );
+    }
```

Comment:
better use ImGui::PushStyleColor( ImGuiCol_ModalWindowDimBg, ImVec4( 1, 0.125f, 0.125f, bgBackUp.w ) );
and always pop if you pushed